### PR TITLE
WEBUI-1853: Allow published documents to be reorganised with the clipboard

### PR DESCRIPTION
**Pull request by narasimhahyland**
Backport of #3382 to `maintenance-3.1.x` (same commit, cherry-picked cleanly). The ticket's `fixVersions` targets `3.1.x`.

## Problem

A document published into a section (a proxy) cannot be moved to another section. The clipboard action is not offered for it, so reorganising a publication area means deleting and republishing the document — which changes its reference and breaks every permanent link pointing at it. A customer hit exactly this: they shared a permanent link, later wanted to file the published document under a new sub-section, and had to republish and then chase down every reference.

See [WEBUI-1853](https://hyland.atlassian.net/browse/WEBUI-1853).

## Root cause

Two independent checks block the flow, and only the first is commonly known:

1. `_isAvailable` in `nuxeo-clipboard-toggle-button.js` and `nuxeo-clipboard-documents-button.js` rejected any proxy. Added by #825 for [NXP-27835](https://hyland.atlassian.net/browse/NXP-27835).
2. `canPaste` in `nuxeo-clipboard.js` requires the clipboard entry's `type` to be an accepted child type of the target. A section only accepts `Section`, while a published `File` proxy reports `type: "File"`, so `COPY`/`MOVE` stay disabled even once the proxy is in the clipboard.

Measured on a live instance, before the fix:

| Paste attempt | `canPaste` |
| --- | --- |
| proxy into a Section (what the user wants) | `false` |
| proxy into a Workspace (what [SUPNXP-27219](https://hyland.atlassian.net/browse/SUPNXP-27219) complained about) | `true` |

So lifting only check 1 — the change suggested on the ticket — would enable the paste nobody asked for and still block the one the customer needs. NXP-27835 originally asked for the subtype check to be relaxed for sections, and that half was never implemented.

## Changes

- `nuxeo-clipboard-toggle-button.js`, `nuxeo-clipboard-documents-button.js`: drop the `isProxy` rejection so a published document can reach the clipboard. Where it may then be pasted is decided by `nuxeo-clipboard`.
- `nuxeo-clipboard.js`: `canPaste` now delegates per entry to `_isAllowedInTarget`. A proxy is accepted only when the target carries the `PublishSpace` facet — the publishing service creates proxies there without consulting accepted child types, so the subtype check cannot meaningfully apply. Every non-proxy document keeps the existing subtype rule.
- `nuxeo-document-storage.js`: persist `isProxy` on the stored entry. The clipboard survives navigation via localStorage and the trimmed entry is all `canPaste` gets back, so the flag has to be stored.

Deliberately preserved boundaries: the clipboard still cannot be used to publish (a regular document is still refused by a section), and a proxy still cannot be pasted outside a publication area.

## Test plan

- `npm run lint` and `npm test` green on this base — 2314 unit tests pass.
- New unit tests: `canPaste` for proxy into publication space, proxy outside it, regular document in/outside a publication space, a mixed selection, and both no-subtypes fallbacks. Storage tests assert `isProxy` round-trips.
- Updated the two unit tests that pinned the old "proxy is not clipboardable" behaviour.
- New functional scenarios in `ftest/features/clipboard.feature`: moving a published document into another section, and the boundary case that it cannot be pasted into a workspace.
- Manually verified on a throwaway container; before/after screenshots and recordings are attached to the ticket.
- Confirmed server-side that `Document.Move` preserves the proxy uid (permanent link keeps resolving) and `Document.Copy` produces another proxy.

## Notes

- `isProxy` is additive in localStorage. Entries stored before an upgrade simply lack the flag and are treated as regular documents, which is the pre-existing behaviour; no migration needed.
- `nuxeo-document-storage` is shared with recents/home, which do not read `isProxy`.

Made with [Cursor](https://cursor.com)

[WEBUI-1853]: https://hyland.atlassian.net/browse/WEBUI-1853?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[NXP-27835]: https://hyland.atlassian.net/browse/NXP-27835?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[SUPNXP-27219]: https://hyland.atlassian.net/browse/SUPNXP-27219?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: elements/nuxeo-clipboard/nuxeo-clipboard.js [hunk 1]
```diff
@@ -268,9 +268,21 @@ Polymer({
...
+    const subtypes = doc.contextParameters && doc.contextParameters.subtypes;
+    return documents.every((entry) => this._isAllowedInTarget(entry, doc, subtypes));
+  },
+
+  /**
+   * A proxy is only accepted in a publication space. The publishing service creates proxies
+   * there without consulting the accepted child types, so a section never lists the published
+   * document's type among its subtypes and the regular subtype check can't apply to it.
+   * Anywhere else, and for any non-proxy document, the target's accepted child types decide.
+   */
```
[the remark this answers is not in the document]
Comment (narasimhahyland) [reply elements/nuxeo-clipboard/nuxeo-clipboard.js:281]: Agreed — the JSDoc now spells out the fallback: the target’s accepted child types decide, "falling back to allowing the paste when the target doesn't expose its subtypes". Applied on both bases (af9aa6f, 80eab5a).

**Review by naveen-konda — APPROVED** [review]

**Review by AnilKumarVanga — APPROVED** [review]

**Review by madhurkulshrestha-hyland — APPROVED** [review]
